apps/api-extractor/src/generators/ApiModelGenerator.ts

```diff
@@ -1087,4 +1130,14 @@ export class ApiModelGenerator {
       }
     }
   }
+
+  private _getFileUrlPath(declaration: ts.Declaration): string {
+    const sourceFile: ts.SourceFile = declaration.getSourceFile();
+    const sourceLocation: ISourceLocation = this._collector.sourceMapper.getSourceLocation({
+      sourceFile,
+      pos: declaration.pos
+    });
+
+    return path.relative(this._collector.extractorConfig.projectFolder, sourceLocation.sourceFilePath);
```

Comment:
@zelliott Here we are computing a filesystem path, but then returning it as a "URL path".  For example, if the source file is called "src/@bad+folder/Example.ts" then at some point it may need to get encoded into something like `src/%40bad%2Bfolder/Example.ts` to avoid trouble in the web server.  

How we encode it (URI? IRI?) perhaps depends on the particular target website.  So I think it maybe makes sense to preserve a "raw" URL-like path here, and document clearly in the API model that the consumer is responsible for URL encoding. And we could demonstrate this by performing such encoding appropriately in API Documenter, at the step where it emits its Markdown hyperlinks.

Reply:
On Windows OS, `path.relative()` will in fact include backslashes instead of slashes, which needs to be addressed here, though.

Reply:
> So I think it maybe makes sense to preserve a "raw" URL-like path here, and document clearly in the API model that the consumer is responsible for URL encoding.

If I understand you correctly, you're suggesting to just update the `SourceLocation.fileUrl` doc comment to clarify that URL encoding is a responsibility of the consumer?

I'm wondering if URL construction altogether should be a responsibility of the consumer. The construction we're currently performing feels a little fragile. See the discussion at https://github.com/microsoft/rushstack/pull/3590#discussion_r974683180.

Reply:
Updated to use `path.posix.relative` instead.